bin/ruby-build

```diff
@@ -1195,13 +1196,36 @@ usage() {
   [ -z "$1" ] || exit "$1"
 }
 
+# list all versions
 list_definitions() {
   { for DEFINITION_DIR in "${RUBY_BUILD_DEFINITIONS[@]}"; do
       [ -d "$DEFINITION_DIR" ] && ls "$DEFINITION_DIR"
     done
   } | sort_versions | uniq
 }
 
+# list only latest stable versions excluding RC, preview, dev and EoL'ed
+list_latest_definitions() {
+  { for DEFINITION_DIR in "${RUBY_BUILD_DEFINITIONS[@]}"; do
+      [ -d "$DEFINITION_DIR" ] && \
+      (
+        cd "$DEFINITION_DIR";
+        grep -L -e warn_eol * | \
+        grep -v -e '-rc[0-9]*$' -e '-preview[0-9]*$' -e '-dev$'
+      )
+    done
+  } | filter_previous_versions | sort_versions | uniq
+}
+
+filter_previous_versions() {
```

Comment:
Maybe this could be called `extract_latest_versions`?

Reply:
+1